### PR TITLE
FIX: update qs version

### PR DESCRIPTION
**Pull request by derek-ho**
Signed-off-by: Derek Ho <dxho@amazon.com>

### Description
[Describe what this change achieves]

### Issues Resolved
CVE-2022-24999

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by mengweieric** [comment]
Did you run cypress tests on 1.3? Seems like we are upgrading from 6.5.2 to 6.10.3 which has a 5 minor versions change.

**Review by mengweieric — DISMISSED** [review]

**Review by ps48 — DISMISSED** [review]

**Review by joshuali925 — APPROVED** [review]

**Comment by derek-ho** [comment]
> Did you run cypress tests on 1.3? Seems like we are upgrading from 6.5.2 to 6.10.3 which has a 5 minor versions change.

@mengweieric a good call out... I have never ran cypress before -> I'm getting a few failures, is that expected? Does that happen even on 2.x? Are cypress tests flaky? We may want to hold off on merging this if not....

**Comment by mengweieric** [comment]
> > Did you run cypress tests on 1.3? Seems like we are upgrading from 6.5.2 to 6.10.3 which has a 5 minor versions change.
> 
> @mengweieric a good call out... I have never ran cypress before -> I'm getting a few failures locally, is that expected? Does that happen even on 2.x? Are cypress tests flaky? We may want to hold off on merging this if not....

Could be flaky tests, what are those failed? 1.3 branch should have all cypress tests passed.

**Comment by derek-ho** [comment]
> > > Did you run cypress tests on 1.3? Seems like we are upgrading from 6.5.2 to 6.10.3 which has a 5 minor versions change.
> > 
> > 
> > @mengweieric a good call out... I have never ran cypress before -> I'm getting a few failures locally, is that expected? Does that happen even on 2.x? Are cypress tests flaky? We may want to hold off on merging this if not....
> 
> Could be flaky tests, what are those failed? 1.3 branch should have all cypress tests passed.

@mengweieric tests are still running... a few examples: 
3) Saves a query on explorer page
       Saves a query on event tab of explorer page:
     AssertionError: Timed out retrying: Expected to find content: 'Mock Flight Events Overview' within the element: <button.euiLink.euiLink--primary> but never did.
      at Context.eval (http://localhost:5601/__cypress/tests?p=.cypress/integration/1_event_analytics.spec.js:289:73)

  4) Saves a query on explorer page
       Saves a visualization on visualization tab of explorer page:
     AssertionError: Timed out retrying: Expected to find content: 'Mock Flight count by destination' within the element: <button.euiLink.euiLink--primary> but never did.
      at Context.eval (http://localhost:5601/__cypress/tests?p=.cypress/integration/1_event_analytics.spec.js:304:73)

  5) Saves a query on explorer page
       Saves a visualization to an existing panel:
     AssertionError: Timed out retrying: Expected to find element: `input[value="Mock Testing Panels"]`, but never found it.
      at Context.eval (http://localhost:5601/__cypress/tests?p=.cypress/integration/1_event_analytics.spec.js:321:8)

Failed 5 for event analytics, 6 for notebooks so far. I can't tell if they're flaky or something changed in the code?

**Comment by rupal-bq** [comment]
you can use qs 6.5.3 for this CVE if. Fix was backported to qs 6.9.7, 6.8.3, 6.7.3, 6.6.1, 6.5.3, 6.4.1, 6.3.3, and 6.2.4
Ref: https://www.cve.org/CVERecord?id=CVE-2022-24999

**Review by rupal-bq — APPROVED** [review]

File: dashboards-observability/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "glob-parent": "^6.0.1",
     "ansi-regex": "^5.0.1",
     "json-schema": "^0.4.0",
+    "qs": "~6.5.3",
```
Comment (joshuali925) [inline dashboards-observability/package.json:43]: i think this is not necessary, `~` doesn't restrict patch version, but it's ok

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**